openfeature-provider/go/confidence/flag_logs_test.go

```diff
@@ -0,0 +1,307 @@
+	err = openfeature.SetProviderAndWait(provider)
+	if err != nil {
+		t.Fatalf("Failed to set provider: %v", err)
+	}
+
+	// Set evaluation context
+	evalCtx := openfeature.NewEvaluationContext(unitTestTargetingKey, map[string]interface{}{
+		"sticky": false,
+	})
+	openfeature.SetEvaluationContext(evalCtx)
+
+	// Clear any logs captured during initialization
+	capturingLogger.Clear()
+
+	client := openfeature.NewClient("flag-logs-test")
+	return capturingLogger, client
+}
+
+// flushAndWait shuts down OpenFeature and waits a bit for async operations
+func flushAndWait() {
+	openfeature.Shutdown()
+	// Give async operations time to complete
+	time.Sleep(100 * time.Millisecond)
+}
+
+func TestFlagLogs_ShouldCaptureWriteFlagLogsAfterBooleanResolve(t *testing.T) {
+	capturingLogger, client := setupFlagLogsUnitTest(t)
+
+	ctx := context.Background()
+
+	// Resolve a boolean flag
+	value, err := client.BooleanValue(ctx, "web-sdk-e2e-flag.bool", true, openfeature.EvaluationContext{})
+	if err != nil {
+		t.Fatalf("Failed to resolve boolean flag: %v", err)
+	}
+	if value != false {
+		t.Errorf("Expected false, got %v", value)
+	}
+
+	// Shutdown to flush logs
+	flushAndWait()
+
+	// Verify captured flag logs
+	requests := capturingLogger.GetCapturedRequests()
+	if len(requests) == 0 {
+		t.Fatal("Expected captured requests, got none")
+	}
+
+	request := requests[0]
+	if len(request.FlagAssigned) == 0 {
+		t.Fatal("Expected flag_assigned entries, got none")
+	}
+
+	// Find the flag we resolved
+	found := false
+	for _, fa := range request.FlagAssigned {
+		for _, af := range fa.Flags {
+			if strings.Contains(af.Flag, "web-sdk-e2e-flag") {
+				found = true
+				if af.TargetingKey != unitTestTargetingKey {
+					t.Errorf("Expected targeting key %s, got %s", unitTestTargetingKey, af.TargetingKey)
+				}
+			}
+		}
+	}
+	if !found {
+		t.Error("Expected to find web-sdk-e2e-flag in captured requests")
+	}
+}
+
+func TestFlagLogs_ShouldCaptureCorrectVariantInFlagLogs(t *testing.T) {
+	capturingLogger, client := setupFlagLogsUnitTest(t)
+
+	ctx := context.Background()
+
+	// Resolve a string flag
+	value, err := client.StringValue(ctx, "web-sdk-e2e-flag.str", "default", openfeature.EvaluationContext{})
+	if err != nil {
+		t.Fatalf("Failed to resolve string flag: %v", err)
+	}
+	if value != "control" {
+		t.Errorf("Expected 'control', got %v", value)
+	}
+
+	// Shutdown to flush logs
+	flushAndWait()
+
+	requests := capturingLogger.GetCapturedRequests()
+	if len(requests) == 0 {
+		t.Fatal("Expected captured requests, got none")
+	}
+
+	request := requests[0]
+	if len(request.FlagAssigned) == 0 {
+		t.Fatal("Expected flag_assigned entries, got none")
+	}
+
+	// Verify variant information is present
+	flagAssigned := request.FlagAssigned[0]
+	if len(flagAssigned.Flags) == 0 {
+		t.Fatal("Expected flags in flag_assigned, got none")
+	}
+
+	appliedFlag := flagAssigned.Flags[0]
+	if appliedFlag.Flag == "" {
+		t.Error("Expected flag name to be non-empty")
+	}
```

Comment:
We could assert a bit more precisely the content of the logs (this applies to other tests in this repo)